### ISSUE TITLE
Fix compile errors

### ISSUE THREAD

**Issue by gskril** [gskril/ens-widgets#13]
When importing the component into another app from the npm package, it throws the following error:

```
TypeError: s.div is not a function
```

And when importing from a local symlink:

```
TypeError: styled_components__WEBPACK_IMPORTED_MODULE_3__.div is not a function
```

Working branch: https://github.com/gskril/ens-widgets/tree/build

**Comment by gskril** [comment]
Turns out this is only the case when server rendering like in Next.js. Works fine in Vite projects 🤔

**Comment by tg44** [comment]
I just run into this, not working on next.js, and we rendering only on the client.

**Comment by gskril** [comment]
@tg44 Is styled-components a dependency or peerDependency in your project? I found making it a regular dependency solves the issue mentioned but then causing a warning when used in another app with styled-components:

```
It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.
```

Which is why I started with it in peerDependencies...

**Comment by tg44** [comment]
We have `"styled-components": "^5.3.6",` in the project.

This solved it, but hacky;
```typescript
      const RegistrationWidget = dynamic(() => import("ens-widgets").then(mod => mod.RegistrationWidget), {
        loading: () => <>Loading...</>,
      })
      return <RegistrationWidget name={widgetProps.ensName}  connectAction={() => {}} wagmiClientConfig={{provider}}/>
```

**Comment by gskril** [comment]
Hmmm yea that works but I'm trying to avoid the need for any extra config on the installation side 🤔. I know it's possible because [family/connectkit](https://github.com/family/connectkit) does it (styled-components is a dependency, but no warnings when using it in another app with styled-components). I've tried modeling my config files close to theirs but it's just not working the same way haha

**Comment by lochie** [comment]
Hey guys, just giving my two cents for how we resolved this over at Family while working on ConnectKit in case this is the same issue we were having when we changed to a pure ES module (worked fine with CommonJS).

styled-components decides to move all of it’s DOM element definitions into `styled.default` rather than just within `styled` definition when building ES module with styled-components as a dependency.

I'm still unsure as to why this occurs but the workaround for this issue that we found over at ConnectKit is to [replace all imports of `styled` with the below utility](https://github.com/family/connectkit/blob/main/packages/connectkit/src/styles/styled/index.ts):
```ts
import { default as s } from 'styled-components';
export const styled = typeof s.div === 'function' ? s : s['default'];
```

We found that other imports from styled-components like `keyframes` and `css` seem to work just fine, only `styled` seems to have this issue.

If you add the above snippet to your [utils.ts](https://github.com/gskril/ens-widgets/blob/main/packages/widget/src/utils.ts), an example of what [ens-widgets/Header.tsx](https://github.com/gskril/ens-widgets/blob/main/packages/widget/src/components/organisms/Header.tsx) might look like is:
```tsx {3}
import { mq, Typography } from '@ensdomains/thorin'
import React from 'react'
import { styled } from './../../utils' // The ESM fix
import { css } from 'styled-components'

import { Logo } from '../atoms/Icons'

const Container = styled.div(
  ({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: ${theme.fontSizes.large};
    ${mq.xs.max(css`
      font-size: ${theme.fontSizes.base};
    `)}
  `
)

const LogoWrapper = styled.div`
  width: 1.5rem;
  svg {
    width: 100%;
    height: 100%;
  }
`

export const Header = () => {
  return (
    <Container>
      <Typography as="label" weight="semiBold" style={{ lineHeight: 1 }}>
        Register an .eth name
      </Typography>
      <LogoWrapper as="a" href="http://ens.domains/" target="_blank">
        <Logo />
      </LogoWrapper>
    </Container>
  )
```

Hope this helps, and if you have any ideas for a better solution please let us know as we'd love to implement a better solution into ConnectKit.

**Comment by gskril** [comment]
Thanks for jumping in @lochie! Unfortunately that doesn't seem to solve the bug - I'm still getting the following error when symlinking the package to a local Next.js project:

```
TypeError: styled_components__WEBPACK_IMPORTED_MODULE_3__.div is not a function
```

The workaround also seems to lose type safety, for example when destructuring `({ theme })` 🥲

I've added a Next.js sample and am running `yarn dev:vite` side by side with `yarn dev:next` and Next is the only issue here. No pressure to try it out obviously, just sharing details here for documentation purposes as I try and figure this out.

**Comment by gskril** [comment]
Closed with #15
